tests/core/test_solve.py

```diff
@@ -1341,14 +1332,14 @@ def test_pinned_1():
             assert convert_to_dist_str(final_state_2) == order
 
         # incompatible CLI and configured specs
-        specs_to_add = MatchSpec("python=2.7"),
+        specs_to_add = MatchSpec("scikit-learn==0.13"),
```

Comment:
Changed to scikit-learn here because the thing we need to test here is whether an indirect conflict with a configured spec will raise.  If you provide an explicit spec, we override any pins.  This is new behavior in this PR.

Elsewhere in index.json, you'll see that all python 2.6 builds of scikit-learn 0.13 have been removed.  That's for this test.  python is the conflict here, but it has to be indirect.